integration/rest/src/test/java/it/gov/pagopa/rtd/transaction_filter/connector/SenderAdeAckRestClientTest.java

```diff
@@ -130,14 +132,47 @@ public void whenDownloadedFileIsEmptyThenDoNotSaveTheFile() {
 
   @SneakyThrows
   @Test
-  public void whenHttpApiReturnsEmptyBodyThenRaisesException() {
+  public void whenDownloadFileIsNotFoundThenDoNotSaveTheFile() {
+    wireMockRule.stubFor(get(urlPathMatching("/ade/.*"))
+        .willReturn(aResponse()));
+
+    List<File> files = restClient.getSenderAdeAckFiles();
+
+    assertThat(files).isNotNull().isEmpty();
+  }
+
+  @SneakyThrows
+  @Test
+  public void whenAdeAckListApiReturnsEmptyBodyThenRaisesException() {
     wireMockRule.stubFor(get("/rtd/file-register/sender-ade-ack")
         .willReturn(aResponse()));
 
     assertThatThrownBy(() -> restClient.getSenderAdeAckFiles()).isInstanceOf(
         NullPointerException.class);
   }
 
+  @SneakyThrows
+  @Test
+  public void whenPutAckReceivedReturns400ThenReturnsNoFilesAndDeleteTempOnes() {
+    wireMockRule.stubFor(put(urlPathMatching("/rtd/file-register/ack-received/.*"))
+        .willReturn(aResponse().withStatus(400)));
```

Comment:
According to internal openAPI we should mock 404, 409 and 503.
The outcome of the test should be the same but we implicitly trace the swagger.

Reply:
It's been utilized the standard validator already unit tested therefore the focus in this test is to verify a value different from 200.